app/Http/Controllers/AuthController.php

```diff
@@ -0,0 +1,235 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Constants\UserType;
+use App\Models\Companies;
+use App\Models\User;
+use App\Repositories\AuthRepository;
+use App\Repositories\CompaniesRepository;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Mail;
+use Illuminate\Support\Facades\Validator;
+
+class AuthController extends Controller
+{
+    /**
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function login(Request $request): JsonResponse
+    {
+        $requestData = $request->all();
+        $validator = Validator::make($requestData, [
+            'email' => 'required|email',
+            'password' => 'required'
+        ]);
+        if ($validator->fails()) {
+            return response()->json(['status' => 500, 'errors' => $validator->errors()]);
+        }
+        $userInfo = User::where('email', $requestData['email'])->first();
+        if (!$userInfo instanceof User) {
+            return response()->json(['status' => 500, 'errors' => ['email' => ['Invalid credential! Please try again']]], 200);
+        }
+        if ($userInfo->activation_code != null) {
+            AuthRepository::sendVerificationEmail($userInfo);
+            return response()->json(['status' => 300, 'message' => 'Your account is not verified yet. we send a mail again. Please verify your account'], 200);
+        } else {
+            if (Hash::check($requestData['password'], $userInfo->password)) {
+                $access_token = $userInfo->createToken('authToken')->accessToken;
+                return response()->json(['status' => 200, 'access_token' => $access_token, 'user' => User::parseData($userInfo)]);
+            }
+        }
+        return response()->json(['status' => 500, 'errors' => ['password' => ['Password is not correct! Please try again.']]], 200);
+    }
+
+    /**
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function socialLogin(Request $request): JsonResponse
+    {
+        $requestData = $request->all();
+        $validator = Validator::make($requestData, [
+            'first_name' => 'required|string',
+            'last_name' => 'nullable|string',
+            'social_provider' => 'required|string',
+            'social_provider_id' => 'required|integer',
+            'email' => 'required|email',
```

Comment:
Email will be nullable. Cause Facebook will not give email to us. google will.
So when email is null then we will generate the email like this 

$email = $requestData->social_provider_id.'@'.$requestData->social_provider.'.com'